Public/Src/IDE/VsCode/BuildXL.IDE.VsCode.dsc

```diff
@@ -1,22 +1,48 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-import {Transformer} from "Sdk.Transformers";
+import {Artifact, Transformer} from "Sdk.Transformers";
 import * as ManagedSdk from "Sdk.Managed";
 import * as Deployment from "Sdk.Deployment";
 import * as Transformers from "Sdk.Transformers";
 import * as DetoursServices from "BuildXL.Sandbox.Windows";
 import * as Branding from "BuildXL.Branding";
 import * as VSIntegration from "BuildXL.Ide.VsIntegration";
+import {Node, Npm} from "Sdk.NodeJs";
 
-namespace LanguageService.Server {
+namespace VsCode.Client {
+    export declare const qualifier: {};
+
+    const clientSealDir = Transformer.sealDirectory(d`client`, globR(d`client`));
+
+    const clientCopy: OpaqueDirectory = Deployment.copyDirectory(clientSealDir.root, Context.getNewOutputDirectory("client-copy"), clientSealDir);
+
+    @public
+    export const installRootDir: OpaqueDirectory = Npm.installFromPackageJson(clientCopy);
+
+    @@public
+    export const compileOutDir: OpaqueDirectory = Npm.runCompile(clientCopy.root, clientCopy, installRootDir);
 
+    @@public
+    export const deployedNpmPackageLockFile = Deployment.copyFileFromOpaqueDirectory(

```

Comment:
TODO: add a comment here about why `Transformer.copyFile` wouldn't work here. 

The comment should be something like: "Here we want to copy a file from inside an opaque output directory.  If we attempted to do that using `Transformer.copyFile` we'd have no way of specifying a dependency of that copy pip to the pip producing this opaque directory, so we wouldn't be able to ensure that the copy operation runs after the opaque directory is produced.